### DISCUSSION TITLE
Why do some Next.js sites ship a manifest including all routes while others don't?

### DISCUSSION THREAD

**Discussion by justincy** [vercel/next.js#14779] [Help] [answered]
I noticed that some sites using Next.js ship a manifest containing info about all routes while other sites don't do this. It's in the `_buildManifest.js` file.

![image](https://user-images.githubusercontent.com/1037458/86289853-a3c9cd00-bba9-11ea-825d-c33a51ddb057.png)

Ship the manifest:

* [Vercel](https://vercel.com/)
* [Nike Help](https://www.nike.com/help)
* [Invision](https://www.invisionapp.com/)

Don't ship the manifest:

* [Hulu](http://hulu.com/)
* [Netflix Jobs](https://jobs.netflix.com/)

What's the difference? Is it an architecture thing (static vs SSG) or is it a version difference? Perhaps the manifest was moved in recent versions and some apps are still using an older version.

I ask because we're building out an admin area but we don't like the idea of making public the existence and location of all our admin routes. If there's a way to disable the manifest then we'd like to do that, otherwise we'll explore how to have two Next.js apps in the same repo so we can share code between our apps.

**Comment by timneutkens** [comment, marked as answer]
> Perhaps the manifest was moved in recent versions and some apps are still using an older version.

It was added as part of https://nextjs.org/blog/next-9-2#improved-code-splitting-strategy given that we need to know what bundles are needed for a route. This has no effect on performance and in general I wouldn't worry about figuring out the routes in that way as there's multiple ways you could figure out that those admin routes exist.

**Reply by stevemarksd** [reply]
Is there any way to hide or remove it? I'm building a multi tenant site and I don't want to show all domains in the manifest.

**Reply by 13luck** [reply]
@stevemarksd
Difficult to hide via customizing next.js config but easy to prevent access with express. Just use `res.sendStatus(204)` while accessing `_buildManifest.js` etc. I know it's so dirty but i'm tire wasting my time 🙇‍♂️. Also:
> the only observed side effect is that clicking these links must be done through a normal `<a href>` instead of the next/link component, which disables client-side navigation
